### DISCUSSION TITLE
Add multiple folders for library

### DISCUSSION THREAD

**Discussion by plato79** [gotson/komga#1524] [Ideas]
### Describe your suggested feature

Currently Komga only supports one folder per library. Is it possible to extend this to scan multiple folders per library? Like Plex does?

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to the latest version.
- [X] I will fill out all of the requested information in this form.

**Comment by gotson** [comment]
What's the use case?

**Comment by plato79** [comment]
I have multiple folders ( from different disks ) where I keep the Mangas. I have to create separate libraries for these right now. Though they all have same configuration ( for Manga ) so I want to keep it as simple as possible.

**Comment by gotson** [comment]
If you are using Docker you could mount them all under the same directory.

**Comment by plato79** [comment]
Though it would be like this then:

/data/Manga/1 ( from disk1 )
/data/Manga/2 ( from disk2 )

Do you suggest then configure library for "/data/Manga" ? Will it successfully scan all the subfolders on both folders?

**Comment by gotson** [comment]
> Do you suggest then configure library for "/data/Manga" ? Will it successfully scan all the subfolders on both folders?

yes, that should work

**Comment by plato79** [comment]
Ugh. That means now I have to change folder structure and that means it'll scan those folders again and it'll take ages ( there are LOTS of manga in my folders ).

You don't suppose there's an easy way to modify folder names of the scanned titles are there ( other than directly via sqlite3 )?

If I have to do it that way, could you suggest which SQL queries I have to run to update folder from "/data/Manga/...." to "/data/Manga/1/...."

**Comment by gotson** [comment]
> Ugh. That means now I have to change folder structure and that means it'll scan those folders again and it'll take ages ( there are LOTS of manga in my folders ).

if you have file hash enabled it should work

> You don't suppose there's an easy way to modify folder names of the scanned titles are there ( other than directly via sqlite3 )?

there's no way to do that, except in DB, which is not supported, so to be done at your own risks

> If I have to do it that way, could you suggest which SQL queries I have to run to update folder from "/data/Manga/...." to "/data/Manga/1/...."

i don't have time to look into what queries would be required, sorry

**Comment by ChocoScaff** [comment]
### Additional context: multiple libraries on multiple disks

I would like to clarify the use case, as it goes beyond having a single library on multiple folders.

I have **multiple libraries** (for example: Manga, Comics, BD), and **each library is split across multiple hard drives** due to storage limitations.

#### Example disk layout

Disk1/media/manga
Disk2/media/manga

Disk1/media/comics
Disk2/media/comics


#### Current limitation
At the moment, Komga only allows **one folder per library**.  
While it is possible to use workarounds such as:
- bind mounts,
- mergerfs,
- artificial root folders,

these solutions add **system-level complexity** and are not always practical, especially for users on non-Linux platforms.

#### Expected behavior
- Allow a library to reference **multiple folders**
- All folders are scanned and managed as a **single logical library**
- Folders could be added or removed independently

#### Why this feature would be useful
- Common scenario for large collections
- Simplifies Docker and multi-disk configurations
- Avoids forcing users to restructure their filesystem
- Improves usability for multi-library setups

I understand there are technical challenges, but I wanted to highlight that this is a real-world use case for users managing large, growing collections across multiple disks.
